src/PrestaShopBundle/Controller/Admin/Sell/Order/CartController.php

```diff
@@ -255,7 +259,56 @@ public function removeCartRuleFromCartAction(int $cartId, int $cartRuleId)
             return $this->json($this->getCartInfo($cartId));
         } catch (Exception $e) {
             return $this->json(
-                ['message' => $this->getErrorMessageForException($e, $this->getErrorMessages())],
+                ['message' => $this->getErrorMessageForException($e, $this->getErrorMessages($e))],
+                Response::HTTP_INTERNAL_SERVER_ERROR
+            );
+        }
+    }
+
+    /**
+     * Adds product to cart
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))")
```

Comment:
Not in the scope of this PR of course, but for API controller actions I think we should have a dedicated Annotation, because the default behaviour of this one is to redirect to the dashboard Returning a json response with 503 code and a quick message would allow us to create those ajax calls more easily